test/new-e2e/containers/eks_test.go

```diff
@@ -0,0 +1,499 @@
+			regexp.MustCompile(`^kube_replica_set:dogstatsd-uds-[[:alnum:]]+$`),
+			regexp.MustCompile(`^pod_name:dogstatsd-uds-[[:alnum:]]+-[[:alnum:]]+$`),
+			regexp.MustCompile(`^pod_phase:running$`),
+			regexp.MustCompile(`^series:`),
+			regexp.MustCompile(`^short_image:apps-dogstatsd$`),
+		},
+	)
+
+	// Test dogstatsd origin detection with UDP
+	suite.testMetric("custom.metric",
+		[]string{
+			"kube_deployment:dogstatsd-udp",
+			"kube_namespace:workload-dogstatsd",
+		},
+		[]*regexp.Regexp{
+			regexp.MustCompile(`^kube_deployment:dogstatsd-udp$`),
+			regexp.MustCompile(`^kube_namespace:workload-dogstatsd$`),
+			regexp.MustCompile(`^kube_ownerref_kind:replicaset$`),
+			regexp.MustCompile(`^kube_ownerref_name:dogstatsd-udp-[[:alnum:]]+$`),
+			regexp.MustCompile(`^kube_qos:Burstable$`),
+			regexp.MustCompile(`^kube_replica_set:dogstatsd-udp-[[:alnum:]]+$`),
+			regexp.MustCompile(`^pod_name:dogstatsd-udp-[[:alnum:]]+-[[:alnum:]]+$`),
+			regexp.MustCompile(`^pod_phase:running$`),
+			regexp.MustCompile(`^series:`),
+		},
+	)
+}
+
+func (suite *eksSuite) TestPrometheus() {
+	// suite.T().Parallel()
+
+	// Test Prometheus check
+	suite.testMetric("prom_gauge",
+		[]string{
+			"kube_deployment:prometheus",
+			"kube_namespace:workload-prometheus",
+		},
+		[]*regexp.Regexp{
+			regexp.MustCompile(`^container_id:`),
+			regexp.MustCompile(`^container_name:prometheus$`),
+			regexp.MustCompile(`^display_container_name:prometheus`),
+			regexp.MustCompile(`^endpoint:http://.*:8080/metrics$`),
+			regexp.MustCompile(`^git.commit.sha:`),                                                       // org.opencontainers.image.revision docker image label
+			regexp.MustCompile(`^git.repository_url:https://github.com/DataDog/test-infra-definitions$`), // org.opencontainers.image.source   docker image label
+			regexp.MustCompile(`^image_id:ghcr.io/datadog/apps-prometheus@sha256:`),
+			regexp.MustCompile(`^image_name:ghcr.io/datadog/apps-prometheus$`),
+			regexp.MustCompile(`^image_tag:main$`),
+			regexp.MustCompile(`^kube_container_name:prometheus$`),
+			regexp.MustCompile(`^kube_deployment:prometheus$`),
+			regexp.MustCompile(`^kube_namespace:workload-prometheus$`),
+			regexp.MustCompile(`^kube_ownerref_kind:replicaset$`),
+			regexp.MustCompile(`^kube_ownerref_name:prometheus-[[:alnum:]]+$`),
+			regexp.MustCompile(`^kube_qos:Burstable$`),
+			regexp.MustCompile(`^kube_replica_set:prometheus-[[:alnum:]]+$`),
+			regexp.MustCompile(`^pod_name:prometheus-[[:alnum:]]+-[[:alnum:]]+$`),
+			regexp.MustCompile(`^pod_phase:running$`),
+			regexp.MustCompile(`^series:`),
+			regexp.MustCompile(`^short_image:apps-prometheus$`),
+		},
+	)
+}
+
+func (suite *eksSuite) testMetric(metricName string, filterTags []string, expectedTags []*regexp.Regexp) {
+	suite.Run(fmt.Sprintf("%s{%s}", metricName, strings.Join(filterTags, ",")), func() {
+		// suite.T().Parallel()
+
+		suite.EventuallyWithTf(func(collect *assert.CollectT) {
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
Oh wow, we might not need to use `backoff.Retry` actually !

Reply:
Yeah. Maybe the `suite.Eventually…` family functions are less flexible regarding the available backoff policies (only constant retry frequency where the `backoff` module implements an exponential backoff policy) but for those tests, that’s enough. And it provides an error collector.